homeassistant/components/motioneye/__init__.py

```diff
@@ -109,13 +149,95 @@ def _add_camera(
 ) -> None:
     """Add a motionEye camera to hass."""
 
-    device_registry.async_get_or_create(
+    def _is_recognized_web_hook(url: str) -> bool:
+        """Determine whether this integration set a web hook."""
+        return f"{WEB_HOOK_SENTINEL_KEY}={WEB_HOOK_SENTINEL_VALUE}" in url
+
+    def _set_webhook(
+        url: str,
+        key_url: str,
+        key_method: str,
+        key_enabled: str,
+        camera: dict[str, Any],
+    ) -> bool:
+        """Set a web hook."""
+        if (
+            entry.options.get(
+                CONF_WEBHOOK_SET_OVERWRITE,
+                DEFAULT_WEBHOOK_SET_OVERWRITE,
+            )
+            or not camera.get(key_url)
+            or _is_recognized_web_hook(camera[key_url])
+        ) and (
+            not camera.get(key_enabled, False)
+            or camera.get(key_method) != KEY_HTTP_METHOD_POST_JSON
+            or camera.get(key_url) != url
+        ):
+            camera[key_enabled] = True
+            camera[key_method] = KEY_HTTP_METHOD_POST_JSON
+            camera[key_url] = url
+            return True
+        return False
+
+    def _build_url(
+        device: dr.DeviceEntry, base: str, event_type: str, keys: list[str]
+    ) -> str:
+        """Build a motionEye webhook URL."""
+
+        return urljoin(
```

Comment:
You can do this nicer with YARL https://yarl.readthedocs.io/en/latest/ . It's already installed as part of HA core.

Reply:
Yeah, I made comment here:
<https://github.com/home-assistant/core/pull/49665#pullrequestreview-668060069>

@dermotduffy Maybe add a comment to that?

Reply:
Good idea. Added a comment explaining why we cannot use YARL.